File: ProyectoMysql/Api/LogisticStorage/LogisticStorage.BusinessLayer/Tarifa.cs
using LogisticStorage.DataLayer;
using LogisticStorage.EntityLayer;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace LogisticStorage.BusinessLayer
{
    public class Tarifa
    {

        public static List<TarifaEntity> ObtenerMain()
        {
            TarifaDB DB = new TarifaDB();
            return DB.ObtenerMain();
        }
        public static List<TarifaEntity> ObtenerItem(Int32 TarifaId)
        {
            TarifaDB DB = new TarifaDB();
            return DB.ObtenerItem(TarifaId);
        }

        public static Int32 Registrar(TarifaEntity Ent)
        {
            TarifaDB DB = new TarifaDB();
            DB.Registrar(Ent);
            return Ent.TarifaId;
        }

        public static List<TarifaEntity> ObtenerMoneda(Int32 MercaderiaId)
        {
            TarifaDB DB = new TarifaDB();
            return DB.ObtenerMoneda(MercaderiaId);
        }

        public static List<TarifaEntity> ObtenerUnidadMedidaPrecio(Int32 MercaderiaId, Int32 MonedaId)
        {
            TarifaDB DB = new TarifaDB();
            return DB.ObtenerUnidadMedidaPrecio(MercaderiaId, MonedaId);
        }
    }
}

File: ProyectoMysql/server/BusinessLayer/Tarifa.py
from DataLayer.TarifaDB import *
from EntityLayer.TarifaEntity import *
from Utilidades.Conexion.configMysql import StartTransaction, EndTransaction, Restore



class Tarifa:
    def ObtenerMain()-> list[TarifaMainModel] :
        try:
            return TarifaDB.ObtenerMain()
        except Exception as e:
            print(e)
    
    def Registrar(Ent: TarifaSaveModel) -> TarifaSaveModel:
        try:
            StartTransaction()
            Value = TarifaDB.RegistrarDB(Ent)
            EndTransaction()
            return Value
        except Exception as e:
            Restore()
            print(e)